### PR TITLE
Fix and test the github admin route

### PR DESCRIPTION
**Pull request by paulmelnikow**
Fix regressions in the github admin and token acceptor endpoints, introduced in #1813.

### CODE REVIEW

File: services/github/auth/admin.spec.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+const { expect } = require('chai')
+const sinon = require('sinon')
+const Camp = require('camp')
+const fetch = require('node-fetch')
+const config = require('../../../lib/test-config')
+const serverSecrets = require('../../../lib/server-secrets')
+const { setRoutes } = require('./admin')
+
+function createAuthHeader({ username, password }) {
```
Comment (paulmelnikow) [inline services/github/auth/admin.spec.js]: It's a bit annoying this must be done manually. We may want to consider a different request library like got or axios.

**Comment by paulmelnikow** [comment]
Sentry issue: [SHIELDS-1B](https://sentry.io/shields/shields/issues/633736321/)

**Comment by PyvesB** [comment]
@paulmelnikow I'll look at this in the morning if no one has done so in the meantime. 😉

**Comment by PyvesB** [comment]
What was the regression that was introduced? Was it the usage of `crypto.timingSafeEqual`? The Sentry logs would probably answser my question, but I don't think I have access to them! 😄

**Comment by paulmelnikow** [comment]
> What was the regression that was introduced? Was it the usage of `crypto.timingSafeEqual`?

Yes it was. That function accepts buffers, not strings, and does not accept `undefined`. There were actually two implementations of `constEq` in the codebase. Probably we could swap out the second for `crypto.timingSafeEqual` if we wrap some tests around `isValidToken`.

> The Sentry logs would probably answser my question, but I don't think I have access to them! 😄 

Indeed! It would be good to give access to more of the maintainers. I wonder if we were to write some guidelines, @espadrine would be open to adding more people.

**Comment by paulmelnikow** [comment]
Thanks for reviewing!

**Pull request merged**